src/Hipace.cpp

```diff
@@ -524,51 +525,49 @@ Hipace::SolveOneSlice (int islice_coarse, const int ibox,
                     "jz", "jz_beam", "rho", "rho_beam", "Psi", "jxx", "jxy", "jyy");
             } else {
                 m_fields.setVal(0., lev, WhichSlice::This,
-                    "ExmBy", "EypBx", "Ez", "Bx", "By", "Bz", "jx", "jx_beam", "jy", "jy_beam",
-                    "jz", "jz_beam", "rho", "rho_beam", "Psi", "jxx", "jxy", "jyy");
+                    "ExmBy", "EypBx", "Ez", "Bx", "By", "Bz", "jx", "jy", "jz", "rho", "Psi");
             }
 
             if (!m_explicit) {
                 m_multi_plasma.AdvanceParticles(m_fields, m_laser, geom[lev], false,
                                                 true, false, false, lev);
             }
 
-            amrex::MultiFab rho(m_fields.getSlices(lev, WhichSlice::This), amrex::make_alias,
-                                Comps[WhichSlice::This]["rho"], 1);
-
             if (m_do_tiling) m_multi_plasma.TileSort(bx, geom[lev]);
             m_multi_plasma.DepositCurrent(
                 m_fields, m_laser, WhichSlice::This, false, true, true, true, m_explicit, geom[lev], lev);
 
             if (m_explicit){
-                m_fields.setVal(0., lev, WhichSlice::Next, "jx", "jx_beam", "jy", "jy_beam");
+                m_fields.setVal(0., lev, WhichSlice::Next, "jx_beam", "jy_beam");
                 m_multi_beam.DepositCurrentSlice(m_fields, geom, lev, islice_local, bins[lev],
                                                  m_box_sorters, ibox, m_do_beam_jx_jy_deposition,
                                                  WhichSlice::Next);
-                m_fields.AddBeamCurrents(lev, WhichSlice::Next);
-                // need to exchange jx jy jx_beam jy_beam
```

Comment:
This AddBeamCurrent is not needed anymore?

Reply:
It actually was never necessary because on `WhichSlice::Next` only `jx_beam` and `jy_beam` are needed with the explicit solver, not `jx` and `jy`.